src/Models/Category.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace Astrogoat\Blog\Models;
+
+use Helix\Fabrick\Icon;
+use Helix\Lego\Models\Contracts\Indexable;
+use Helix\Lego\Models\Contracts\Metafieldable;
+use Helix\Lego\Models\Contracts\Searchable;
+use Helix\Lego\Models\Contracts\Sectionable;
+use Helix\Lego\Models\Model as LegoModel;
+use Helix\Lego\Models\Traits\HasMetafields;
+use Helix\Lego\Models\Traits\HasSections;
+use Spatie\Sluggable\HasSlug;
+use Spatie\Sluggable\SlugOptions;
+
+class Category extends LegoModel implements Sectionable, Indexable, Searchable, Metafieldable
+{
+    use HasSections;
+    use HasSlug;
+    use HasMetafields;
+
+    protected $table = 'blog_categories';
+
+    public static function icon(): string
+    {
+        return Icon::COLLECTION;
+    }
+
+    public function editorShowViewRoute(string $layout = null): string
+    {
+        return route('lego.blog.categories.editor', [
+            'category' => $this,
+            'editor_view' => 'show',
+            'layout' => $layout,
+        ]);
+    }
+
+    public static function getDisplayKeyName(): string
+    {
+        return 'name';
+    }
+
+    public function getSlugOptions(): SlugOptions
+    {
+        return SlugOptions::create()
+            ->generateSlugsFrom($this->getDisplayKeyName())
+            ->saveSlugsTo('slug')
+            ->doNotGenerateSlugsOnUpdate();
+    }
+
+    public function shouldIndex(): bool
+    {
+        return $this->indexable;
+    }
+
+    public function getIndexedRoute(): string
+    {
+        return $this->getPublishedRoute();
+    }
+
+    public function getPublishedAtKey(): string
+    {
+        return 'published_at';
+    }
+
+    public function isVisible(): bool
+    {
+        // TODO: Implement isVisible() method.
+    }
+
+    public function hasBeenPublished(): bool
+    {
+        // TODO: Implement hasBeenPublished() method.
+    }
+
+    public function publishedState(): string
+    {
+        // TODO: Implement publishedState() method.
+    }
+
+    public function getPublishedRoute(): string
+    {
+        return route('lego.blog.categories.show', $this);
+    }
```

Comment:
Since it's not implementing the `Publishable` interface you should be able to get rid of the methods.